bftengine/src/bftengine/ControllerWithSimpleHistory.cpp

```diff
@@ -87,23 +90,20 @@ bool ControllerWithSimpleHistory::onNewSeqNumberExecution(SeqNum n) {
   SeqNoInfo& s = recentActivity.get(n);
 
   Time now = getMonotonicTime();
+  auto duration = now - s.prePrepareTime;
 
-  uint64_t duration = 0;
-  if (now > s.prePrepareTime) duration = subtract(now, s.prePrepareTime);
-
-  const uint64_t MAX_DURATION_MICRO = 2 * 1000 * 1000;  // 2 sec
-  const uint64_t MIN_DURATION_MICRO = 100;              // 100 micro
+  const auto MAX_DURATION_MICRO = microseconds(2 * 1000 * 1000);
+  const auto MIN_DURATION_MICRO = microseconds(100);
 
   if (duration > MAX_DURATION_MICRO)
-    duration = MAX_DURATION_MICRO;
+    s.durationMicro_ = MAX_DURATION_MICRO;
```

Comment:
I think the idea here is to cap "duration" up to MAX_DURATION_MICRO.
Why are we setting s.durationMicro_ instead?

Reply:
duration is a local that we only use past this point to set s.durationMicro_. see line 102. I just removed the setting of the temporary.